src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2014,9 +2028,21 @@ internal void MarkStaticConstructorVisibleToReflection (TypeDefinition type, in
 			if (CheckProcessed (type))
 				return type;
 
+			if (type.HasMethods) {
```

Comment:
Originally I moved it up because it relied on the DependencyInfo, but I think that can condition can be removed and moved back to ProcessMethod. I think the check on the DependencyKind was only to prevent infinite recursion when marking cctors, which we won't hit with the dependency analyzer.